### PR TITLE
feat(TabBar): support TabBar

### PR DESCRIPTION
**Pull request by judastheo**
## Summary
Support `Tab` and `TabBar` components.
Handling the border at TabBar level to ensure correct UI when the border color has different opacities.

## Checklist before requesting a review

- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [ ] Design validated

## Screenshots

| iOS                                                                                                                                                              | Android                                                                                                                                                              |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
|  |  |

### CODE REVIEW

File: src/components/tabs/tab-bar/tab-bar.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        backgroundColor={borderConfig.backgroundColor}
+        height={1}
+        position="absolute"
+        bottom={0}
+        right={0}
+        style={borderStyle}
+        opacity={borderConfig.opacity}
+        alignSelf="center"
+      />
+      <ScrollView
```
Comment (hcourthias) [inline src/components/tabs/tab-bar/tab-bar.tsx]: Use `ScrollView` from `reanimated`

File: src/components/tabs/tab-bar/tab-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  appearance,
+  onTabSelected,
+  tabs,
+  tabIndex,
+}: TabBarProps) => {
+  const { spacing } = useStreamlineTheme();
+  const borderConfig = getBorderConfig({ appearance });
+  const scrollOffset = useSharedValue(0);
+
+  const onScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
```
Comment (hcourthias) [inline src/components/tabs/tab-bar/tab-bar.tsx]: use `useAnimatedScrollHandler` from `reanimated`

File: src/components/tabs/tab/tab.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    <Pressable
+      accessibilityRole="tab"
+      accessibilityLabel={title}
+      onPressIn={onPressIn}
+      onPress={onPress}
+      onPressOut={onPressOut}
+      selected={isSelected}
+    >
+      <Box paddingBottom="md" flex={1}>
+        <AnimatedBox style={textStyle}>
```
Comment (hcourthias) [inline src/components/tabs/tab/tab.tsx:61]: It's fine but I would have prefer a unique view that slide to the active tab (e.g Twitter tabs)

File: src/components/tabs/tab/tab.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            color={
+              isSelected ? titleConfig.selectedColor : titleConfig.defaultColor
+            }
+            opacity={isSelected ? 1 : titleConfig.defaultOpacity}
+            variant="bodyBold"
+          >
+            {title}
+          </Text>
+        </AnimatedBox>
+        <AnimatedBox
```
Comment (hcourthias) [inline src/components/tabs/tab/tab.tsx:72]: It's fine but I would have prefer a unique view that slide to the active tab (e.g Twitter tabs)

**Review by ianrobertlambert — APPROVED** [review]
Looks good to me! Thanks :)
For the demo, can we please make sure the `TabBar` spans the entire width of the screen?

**Comment by judastheo** [comment]
> Looks good to me! Thanks :) For the demo, can we please make sure the `TabBar` spans the entire width of the screen?

@ianrobertlambert You mean an example where the Tabs overflow the width of the devices right?

**Comment by ianrobertlambert** [comment]
> > Looks good to me! Thanks :) For the demo, can we please make sure the `TabBar` spans the entire width of the screen?
> 
> @ianrobertlambert You mean an example where the Tabs overflow the width of the devices right?

Yes please! And, can we please align the TabBar to the left and right edge of the screen? Like this:

image

**Comment by judastheo** [comment]
@ianrobertlambert  Updated with the minor fix and adding the example!

**Review by hcourthias — APPROVED** [review]

**Pull request merged**